napari/_qt/widgets/qt_tooltip.py

```diff
@@ -13,3 +13,23 @@ def enterEvent(self, event):
             QToolTip.showText(pos, self.toolTip(), self)
 
         super().enterEvent(event)
+
+
+class QtTopToolTipLabel(QLabel):
+    """
+    A QLabel that provides instant tooltips on mouser hover positioned over the label top.
+    """
+
+    def enterEvent(self, event):
+        """
+        Override to show tooltips instantly and set their position over the label top.
+        """
+        if self.toolTip():
+            point = self.contentsRect().center()
+            y_offset = self.contentsRect().height() * 3
+            new_y = int(point.y() - y_offset)
+            point.setY(new_y)
+            pos = self.mapToGlobal(point)
+            QToolTip.showText(pos, self.toolTip(), self)
+
+        super().enterEvent(event)
```

Comment:
See https://github.com/napari/napari/blob/main/napari/_qt/widgets/qt_tooltip.py

Reply:
Yep, that's why I added this new subclass of `QLabel` there :+1:

The only difference is that this new subclass `QtTopToolTipLabel` puts the tooltip vertically on the oposite direction than the existing class (to prevent the tooltip from showing outside the window and also follow the designs at #5326):

* Using the class I added here (`QtTopToolTipLabel`) you get:

![top_tooltip](https://user-images.githubusercontent.com/16781833/204040547-26b00ad7-e5e0-4b7a-a2dd-6c49bb5fbf2c.png)

* Using the already available class (`QtToolTipLabel`) you get:

![bottom_tooltip](https://user-images.githubusercontent.com/16781833/204040861-7201e696-4065-4edd-afd2-79508d445ad5.png)

Reply:
Could we add a parameter to account for this? Or reuse it s bit?

Reply:
Will check :+1: Although taking into account further feedback, maybe using already available functionality to display the message could be better?